docs/reference/contrib/modeladmin/reversing_urls.rst

```diff
@@ -0,0 +1,126 @@
+.. _modeladmin_reversing_urls:
+
+=========================
+Reversing ModelAdmin URLs
+=========================
+
+It's sometimes useful to be able to derive the ``index`` (listing) or
+``create`` URLs for a model along with the ``edit``, ``delete`` or
+``inspect`` URL for a specific object in a model you have registered via
+the ``modeladmin`` app.
+
+Wagtail itself does this by instantiating of each ``ModelAdmin`` class you have
+registered, and using the ``url_helper`` attribute of each instance to
+determine what these URLs are.
+
+You can take a similar approach in your own code too, by creating a
+``ModelAdmin`` instance yourself, and using its ``url_helper``
+to determine URLs.
+
+See below for some examples:
+
+.. contents::
+    :local:
+    :depth: 1
+
+-------------------------------------------------------------------
+Getting the ``edit`` or ``delete`` or ``inspect`` URL for an object
+-------------------------------------------------------------------
+
+In this example, we will provide a quick way to ``edit`` the Author that is
+linked to a blog post from the Admin Page Listing menu. We have defined
+an ``AuthorModelAdmin`` class and registered it with Wagtail to allow
+``Author`` objects to administered via the admin area. The ``BlogPage`` model
+has an ``author`` field (a ``ForeignKey`` to the ``Author`` model) to allow a
+single author to be specified for each post.
+
+.. code-block:: python
+
+    # file: wagtail_hooks.py
+
+    from wagtail.admin.widgets import PageListingButton
+    from wagtail.contrib.modeladmin.options import ModelAdmin, modeladmin_register
+    from wagtail.core import hooks
+
+    # Authors & BlogPage model not shown in this example
+    from models import Authors
+
+    # ensure our modeladmin is created
+    class AuthorModelAdmin(ModelAdmin):
+        model = Authors
+        menu_order = 200
+
+    # Creating an instance of `AuthorModelAdmin`
+    author_modeladmin = AuthorModelAdmin()
+
+    @hooks.register('register_page_listing_buttons')
+    def page_listing_buttons(page, page_perms, is_parent=False):
```

Comment:
How do you feel about changing the name of this method to `add_author_edit_buttons`?

Reply:
sweet done